### PR TITLE
bedops: import from homebrew/science

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Companion PR: Homebrew/homebrew-science#6598
Cross reference: Homebrew/homebrew-science#6331

### CODE REVIEW

File: Formula/bedops.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
+class Bedops < Formula
+  desc "Set and statistical operations on genomic data of arbitrary scale"
+  homepage "https://github.com/bedops/bedops"
+  url "https://github.com/bedops/bedops/archive/v2.4.30.tar.gz"
+  sha256 "218e0e367aa79747b2f90341d640776eea17befc0fdc35b0cec3c6184098d462"
+
+  head "https://github.com/bedops/bedops.git"
```
Comment (ilovezfs) [inline Formula/bedops.rb]: please remove head

File: Formula/bedops.rb [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+  needs :cxx11
+
+  def install
+    ENV.O3
```
Comment (ilovezfs) [inline Formula/bedops.rb]: why? The default is -Os which should be used unless there's a bug

File: Formula/bedops.rb [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    system "make"
```
Comment (ilovezfs) [inline Formula/bedops.rb]: Is the separate `make` needed or is `make install` sufficient?

File: Formula/bedops.rb [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    system "make", "install"
+    bin.install Dir["bin/*"]
```
Comment (ilovezfs) [inline Formula/bedops.rb]: just pass `BINDIR=#{bin}` on the `make install` line

File: Formula/bedops.rb [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+  test do
+    (testpath/"first.bed").write <<~EOS
+      chr1	100	200
+    EOS
+    (testpath/"second.bed").write <<~EOS
+      chr1	300	400
+    EOS
+    assert_match "chr1	200	300",
+      shell_output("#{bin}/bedops --complement first.bed second.bed")
```
Comment (ilovezfs) [inline Formula/bedops.rb]:
``` 
      output = shell_output("#{bin}/bedops --complement first.bed second.bed")
      assert_match "chr1\t200\t300", output
```

File: Formula/bedops.rb [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  def install
+    ENV.O3
+    system "make"
+    system "make", "install"
+    bin.install Dir["bin/*"]
+  end
+
+  test do
+    (testpath/"first.bed").write <<~EOS
+      chr1	100	200
```
Comment (ilovezfs) [inline Formula/bedops.rb]: use \t

File: Formula/bedops.rb [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+    EOS
+    (testpath/"second.bed").write <<~EOS
+      chr1	300	400
```
Comment (ilovezfs) [inline Formula/bedops.rb]: use \t

**Review by ilovezfs — CHANGES_REQUESTED** [review]

File: Formula/bedops.rb (hunk 3 above)
Comment (EricFromCanada) [reply Formula/bedops.rb]: Separate steps are required.

File: Formula/bedops.rb (hunk 2 above)
Comment (EricFromCanada) [reply Formula/bedops.rb]: Long ago in Homebrew/homebrew-science#726 it was deemed necessary, but no explanation was given. Seems to work fine without it.

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
Thanks again @EricFromCanada!

**Pull request closed without merging**